esmvalcore/config/extra_facets/native6-mappings.yml

```diff
@@ -0,0 +1,192 @@
+# Extra facets for native6 data
+
+# Notes:
+# - All facets can also be specified in the recipes. The values given here are
+#   only defaults.
+
+# A complete list of supported keys is given in the documentation (see
+# ESMValCore/doc/quickstart/find_data.rst).
+---
+
+ERA5:
+
+  # Settings for all variables of all MIPs
+  '*':
+    '*':
+      family: E5
+      regrid: true
+      type: an
+      typeid: '00'
+      version: ''  # necessary to get a nice output file name
+
+    # Variable-specific settings
+    albsn:
```

Comment:
Would it be possible to add `var_name` or a similar name (e.g. standard_name, long_name) to this mapping and add a simple check in `AllVars.fix_metadata` that the input data actually contains the right variable? I've always been concerned that people may put the wrong file in the wrong directory and get completely wrong results out of these fixes.

Reply:
In principle I like that idea, but I don't know if that works in practice. For example, are we sure that the names of the netCDF input files are the exact same as for the GRIB input files? If this is not the case, the fix will start failing for one or the other.

I would argue that for the GRIB files the risk of using wrong files is rather low, as the GRIB ID (= the variable name) is [part of the file name](https://github.com/ESMValGroup/ESMValCore/blob/cc632dc4aed304d8751f200e729bf08c59da414b/esmvalcore/config-developer.yml#L105). So this is more of a problem for the netCDF input files, for which we simple use `*.nc` as input file pattern.

Reply:
I would expect (but have not checked), that ECMWF uses the same names for their variables, regardless of the file format. e.g. are these variable names present in both file formats? https://confluence.ecmwf.int/pages/viewpage.action?pageId=82870405#ERA5:datadocumentation-Table4

Reply:
While this is probably true, I don't think this check is relevant for the GRIB files. As I mentioned, they have their GRIB ID (= variable name) in the file name. We also don't check the variable name for CMIP model output.

Thus, IMHO, it would make more sense to implement this in a different PR (this one here as already >1000 lines).

Reply:
> We also don't check the variable name for CMIP model output.

That is exactly what the CMOR checks do, right? In this case, we're overwriting the variable name as part of the fix, regardless of what was there. As long as people stick with using the variables that you have pre-defined for them in this pull request, that seems quite safe, but what if they would start adding others? Or did you add support for all the ERA5 data available on DRKZ?

Reply:
> That is exactly what the CMOR checks do, right?

No, the CMOR checks don't do that: https://github.com/ESMValGroup/ESMValCore/issues/1907. They will simply overwrite wrong long names and standard names (regardless of what was there), but wrong variables names are simply ignored.

Sorry, I don't get your point. What would checking the variable name help if people started to use (unsupported) variables? We could of course add a list of supported variable names and raise errors for others, but just checking against a predefined value given in the extra facets won't help at all (as you said, users could simply overwrite that in the recipe).

For GRIB, I really don't see the point in adding this check. For all supported variables, a GRIB ID is predefined, and for all others, users need to give that in the recipe. And if users **explicitly** give that in the recipe, I think we should be fine with that and assume they know what they're doing.

Reply:
My main concern is that if users download data, either as NetCDF or as GRIB, there is a risk that they make a mistake and put a file containing the wrong variable in the directory with the variable name or GRIB ID that we use to look up the files corresponding to a particular variable. Since we are overwriting all variable names in the fix, this may go unnoticed. That is why it would make sense to me to check that the input variable name matches with what we expect before overwriting.

A secondary concern is that it may not be clear to users which variables are supported and which are not. This is probably a bigger issue with NetCDF input than with GRIB files though, since you added a nice list of supported in GRIB [here](https://github.com/ESMValGroup/ESMValCore/blob/read_era5_grib/esmvalcore/config/extra_facets/native6-era5.yml).

> No, the CMOR checks don't do that: https://github.com/ESMValGroup/ESMValCore/issues/1907. They will simply overwrite wrong long names and standard names (regardless of what was there), but wrong variables names are simply ignored.

You're right, that does not look safe at all.

Reply:
Ok, just to make sure I understand you correctly: you propose to add a new facet (something like `raw_name`) to the extra facets table and check the variables name of the raw file against that, right?

I fully agree that this would make sense for the netcdf files. They currently use `*.nc` as input file pattern, so basically every netcdf file in the input directory is read. This will lead to various problems if users put wrong files into the input directories.

However, for ERA5 GRIB files, I think the situation is a lot different:

1. For the data pool available on DKRZ, the input file convention is `'{family}{level}{typeid}_{tres}_*_{grib_id}.grb'`. Since the GRIB ID is basically the variable name, there's no way of selecting a wrong file. Also, users won't need to copy around files themselves. If they do that, they're on their own (we have no way of making sure everything works fine if users start modifying and renaming files; but this can happen for any data, not only ERA5 GRIB).
2. For self-downloaded data, we specifically mention that this is *experimental* and users need additional setup. I don't know what the naming conventions for self-downloaded data are (I don't even know if they are consistent across the [various ERA5 products](https://cds.climate.copernicus.eu/datasets) you can download), so I am really hesitant to enforce any additional restrictions on the data without more tests and an in-depth analysis.

In summary: I think checking variables names is a good idea for the netcdf files and maybe for self-downloaded GRIB files, but this should go into a separate PR. The main point of this PR is to add support for the ERA5 data pool that is available on Levante, and I am convinced that there's no risk of using wrong files if users don't start modifying/renaming/copying files themselves.

Reply:
I expect users will be quick to rebuild the DKRZ data structure on their own computers if they don't have access to DKRZ. But you're right that having the grib_id in the filename provides some safety, so I'm happy to leave this feature for a future pull request. See also https://github.com/ESMValGroup/ESMValCore/pull/1125 for some previous discussion on the topic.